internal/authorization/helpers.go

```diff
@@ -0,0 +1,19 @@
+package authorization
+
+import (
+	"context"
+)
+
+type ClientAuthorizer = authorizer
```

Comment:
I'm not a big fan of this approach for two reasons:
- if you use it as a new object you need to initiate it, but then you would initiate an entire authorizer just for the helper methods
- if you use this starting from an existing object, to have these method available then you would need to cast the existing object to `ClientAuthorizer`

My opinion on this is that it makes things unnecessarily more complicated than they need to be.
For example the `ClientResource` method could be an exported utility function since it's very useful
Meanwhile the `CreateClient` seems to belong the main Authorizer struct.